imports/models/BoardFavoritesCollection.ts

```diff
@@ -0,0 +1,55 @@
+import { BaseCollection } from './BaseCollection'
+import { BoardRecord } from './BoardCollection'
+
+export interface BoardFavoriteRecord {
+  userid: string
+  boardId: string
+}
+
+export class BoardFavoritesCollection extends BaseCollection<BoardFavoriteRecord> {
+  constructor() {
+    super('board_favorites')
+  }
+
+  public async addFavoriteBoard(userid: string, boardId: string): Promise<BoardFavoriteRecord> {
+    const query = { userid, boardId }
+    const update = { $set: { userid, boardId } }
+    const result = await this.collection.upsertAsync(query, update)
+    if (result && result.numberAffected && result.numberAffected > 0) {
+      const newRecord = { userid, boardId }
+      return newRecord
+    } else if (result && result.numberAffected === 1) {
```

Comment:
looks werid here. `result?.numberAffected > 0` already contains `result?.numberAffected === 1` scenario, so this branch will never be jumpped into?

Reply:
yeah I wanted your clarification on it

Reply:
if the result's data structure is like { numberAffected?: number }

we can simply `return result?.numberAffected > 0`, to make sure signature is same as `removeFavoriteBoard`